### ISSUE TITLE
resolution of relational operators

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#22430]
I'm not sure how routines should be written so both of these give the same answer:
```python
>>> var('x',real=1)
x
>>> floor(x)<=x
True
>>> x>=floor(x)
x >= floor(x)
```
I don't think each routine should try reverse args otherwise there would be a recursion error. What is the right way to handle this so regardless of how the relational is expressed it evaluates?

**Comment by ThePauliPrinciple** [comment]
I think if one of them gives returns `NotImplemented` then the other object's respective comperative magic method is used as a back up.

**Comment by oscarbenjamin** [comment]
Nothing should be handled in the `__le__` methods directly. Those should pass of to `is_le`.

**Comment by smichr** [comment]
> Those should pass off to `is_le`.

So the goal is that all the relational `__op__` methods are just stubs to call `is_le`?

**Comment by oscarbenjamin** [comment]
> So the goal is that all the relational `__op__` methods are just stubs to call `is_le`?

Yes. There should be no special case logic and everything should pass through a single routine for consistency.
